golem-service-base/src/worker_executor_clients.rs

```diff
@@ -1,25 +1,50 @@
+use crate::model::{GolemError, GolemErrorUnknown, Pod};
 use async_trait::async_trait;
 use golem_api_grpc::proto::golem::workerexecutor::worker_executor_client::WorkerExecutorClient;
+use golem_common::cache::{BackgroundEvictionMode, Cache, FullCacheEvictionMode, SimpleCache};
+use std::time::Duration;
 use tonic::transport::Channel;
 
-use crate::model::Pod;
-
 #[async_trait]
 pub trait WorkerExecutorClients {
-    async fn lookup(&self, pod: &Pod) -> Result<WorkerExecutorClient<Channel>, String>;
+    async fn lookup(&self, pod: Pod) -> Result<WorkerExecutorClient<Channel>, GolemError>;
+}
+
+pub struct WorkerExecutorClientsDefault {
+    cache: Cache<Pod, (), WorkerExecutorClient<Channel>, GolemError>,
 }
 
-#[derive(Default)]
-pub struct WorkerExecutorClientsDefault {}
+impl WorkerExecutorClientsDefault {
+    pub fn new(max_capacity: usize, time_to_idle: Duration) -> WorkerExecutorClientsDefault {
+        WorkerExecutorClientsDefault {
+            cache: Cache::new(
+                Some(max_capacity),
+                FullCacheEvictionMode::LeastRecentlyUsed(1),
+                BackgroundEvictionMode::OlderThan {
+                    ttl: time_to_idle,
+                    period: Duration::from_secs(60),
+                },
+                "worker_connect_client",
```

Comment:
This name is a bit misleading, these clients are not only used for the `worker connect` feature, but for all gRPC calls to the worker executors

Reply:
True, worker_grpc_client may be?